### ISSUE TITLE
Readd :shell command

### ISSUE THREAD

**Issue by ZyX-I** [neovim/neovim#3439]
Now with the `:terminal` merged I think it makes sense to readd the `:shell` command which will basically do the following:

1. Open new tab.
2. Save `&showtabline` and `&showstatus` variables and set them to zero.
3. Run `:terminal`.
4. Add an autocmd which restores `&show*` options when terminal exits.
5. Add something which make it exit immediately in place of showing “[Program exited, press any key to close]”.

\: simple, easy (except that I do not know what to do with 5) and makes NeoVim more compatible.

**Comment by tarruda** [comment]
@ZyX-I yes, this is simple/elegant way to emulate `:shell`, but it will still be asynchronous. It might be possible to combine this with `jobwait` to create a synchronous version though. 

As for 5, perhaps the `on_exit` callback can be used? This is how FZF closes the terminal after the user presses enter.

**Comment by ZyX-I** [comment]
@tarruda If making it synchronous is required it is enough to just disable `<C-\><C-n>` as it is the only way to exit terminal mode.

**Comment by ZyX-I** [comment]
Though no, I guess that API calls can be used to do the same thing, and also switch buffers/windows/whatever. But is this really required: some other features like `!` are also not fully compatible.

Not sure I understood what you meant by “synchronous version” correctly though.

**Comment by tarruda** [comment]
> @tarruda If making it synchronous from the user point of view is required it is enough to just disable <C-\><C-n> as it is the only way to exit terminal mode.

No, see [the implementation](https://github.com/neovim/neovim/blob/master/src/nvim/ex_docmd.c#L9420-L9423). It uses `termopen()` and `:startinsert` which means nvim will first finish processing all commands and only when it restarts the normal mode command loop it will enter terminal mode.

By synchronous I meant a version of `:shell` that only returns when the program exits(Unless `:shell` is already asynchronous, if so then nevermind)

**Comment by ZyX-I** [comment]
Now I understand. Unable to use VimL to script synchronous variant: `jobwait()` can only be used with some C code hacks because input processing will not be done in this case, neither it will update the shell window.
